src/Microsoft.NetFramework.Analyzers/Core/SerializationRulesDiagnosticAnalyzer.cs

```diff
@@ -264,34 +266,20 @@ private bool IsSerializable(ITypeSymbol type)
                         return true;
 
                     default:
-                        return IsPrimitiveType(type) ||
-                            type.SpecialType == SpecialType.System_String ||
-                            type.SpecialType == SpecialType.System_Decimal ||
-                            type.GetAttributes()
-                                .Any(a => a.AttributeClass.Equals(_serializableAttributeTypeSymbol));
-                }
-            }
+                        if (type.GetAttributes().Any(a => a.AttributeClass.Equals(_serializableAttributeTypeSymbol)))
+                        {
+                            return true;
+                        }
+
+                        // Workaround for https://github.com/dotnet/roslyn/issues/3898
+                        var isSerializableMetadataProperty = type.GetType().GetRuntimeProperties().FirstOrDefault(p => p.Name.Equals("IsSerializable", StringComparison.Ordinal));
+                        if (isSerializableMetadataProperty != null &&
+                            isSerializableMetadataProperty.GetValue(type) is bool isSerializable &&
+                            isSerializable)
```

Comment:
nit: you can put this condition into the Where clause. Then, you do not need to extract the FirstOrDefault instance.